Gem/Swap/Types/SwapQuoteInput.swift

```diff
@@ -2,15 +2,41 @@
 
 import Foundation
 import Primitives
+import BigInt
 
 struct SwapQuoteInput: Hashable, Sendable {
-    let fromAsset: AssetData?
-    let toAsset: AssetData?
-    let amount: String
+    let fromAsset: Asset
+    let toAsset: Asset
+    let amount: BigInt
 }
 
 // MARK: - Identifiable
 
 extension SwapQuoteInput: Identifiable {
-    var id: String {"\(fromAsset?.id ?? "")_\(toAsset?.id ?? "")_\(amount)" }
+    var id: String {"\(fromAsset.id)_\(toAsset.id)_\(amount)" }
```

Comment:
Not must have, just syntaxes sugar suggestion

```suggestion
var id: String { [fromAsset?.id ?? "", toAsset?.id ?? "", amount].joined(separator: "_") }
```

or

```suggestion
var id: String { [fromAsset?.id, toAsset?.id, amount].compactMap { $0 }.joined(separator: "_") }
```